voluncheer/chatroom/consumers.py

```diff
@@ -33,16 +58,22 @@ def disconnect(self, close_code):
             self.channel_name,
         )
 
-    def receive(self, text_data=None, bytes_data=None):
-        text_data_json = json.loads(text_data)
-        message = text_data_json["message"]
-
+    def receive(self, text_data, bytes_data=None):
+        test_data_json = json.loads(text_data)
+        print(test_data_json)
+        content = test_data_json["message"]
+        user = test_data_json["user"]
+        room = test_data_json["room"]
+        timestamp = test_data_json["timestamp"]
+        photo = test_data_json["photo"]
+        self.save_message(user, room, content, timestamp)
         # send chat message event to the room
         async_to_sync(self.channel_layer.group_send)(
             self.room_group_name,
             {
                 "type": "chat_message",
-                "message": message,
+                "message": content,
```

Comment:
Content is a field of message. Why are we assigning it to "message"? is it called message in html?

Reply:
Message class has fields: user, room, content, timestamp, volunteer, organization. So I am a bit confused why we are assigning content to message and calling message.content in room.html